donkeycar/management/makemovie.py

```diff
@@ -236,19 +236,18 @@ def make_frame(self, t):
 
         if self.do_salient:
             image = self.draw_salient(image)
-            image = image * 255
-            image = image.astype('uint8')
-        
+            image = cv2.normalize(src=image, dst=None, alpha=0, beta=255, norm_type=cv2.NORM_MINMAX, dtype=cv2.CV_8U)
```

Comment:
Just a comment: I don't think there is a material difference between upscaling the [0, 1.0] image to [0, 255] integer, except probably the rounding.

Reply:
Unfortunately some pixels of the `image` contain RGB values over 1.0 due the blending of `cv2.addWeighted` https://github.com/autorope/donkeycar/blob/20bf83c4f8bfe24bda151659e271a488bfa91f1b/donkeycar/management/makemovie.py#L219-L220
With the original approach, https://github.com/autorope/donkeycar/blob/20bf83c4f8bfe24bda151659e271a488bfa91f1b/donkeycar/management/makemovie.py#L239-L240
the multiplication by 255 produces pixels with values over 255. Then the values are modulo 255 under the uint8 conversion of numpy (`image.astype('uint8')`). For example, a float value of 1.1 is multipled to (1.2*255 = 306), then it is converted to a uint8 value of (306 mod 255 = 51). It is better to use `cv2.normalize` since it can always normalize the pixels' RGB correctly regardless of whether some pixels exceed the range.

Reply:
For example, a float value of 1.2 is multipled to (1.2*255 = 306), then it is converted to a uint8 value of (306 mod 255 = 51).

**typo

Reply:
Makes sense, thanks for explaining.